Rude/Rude.cs

```diff
@@ -8,208 +8,77 @@
 
 namespace Rude
 {
-    public static class LinqExtension
+    public class RudePlugin : PluginBase
     {
-        public static double? Median<TColl, TValue>(
-            this IEnumerable<TColl> source,
-            Func<TColl, TValue>     selector)
-        {
-            return source.Select<TColl, TValue>(selector).Median();
-        }
-
-        public static double? Median<T>(
-            this IEnumerable<T> source)
-        {
-            if(Nullable.GetUnderlyingType(typeof(T)) != null)
-                source = source.Where(x => x != null);
-
-            int count = source.Count();
-            if(count == 0)
-                return null;
+        private readonly string[] Prefixes = { "[Rude] ", "[Ruderer] ", "[Rüdiger] " };
+        public static readonly TimeSpan RudeCooldown = TimeSpan.FromMinutes(30);
```

Comment:
FYI the concept of cooldowns in the previous implementation meant that a given user could only rude once every 15 mins or so. This is IMO a **very** desirable feature as it prevents you from doing things like instakicking somebody or mass-ruding everyone on the server.